frontend/app/components/data-table.tsx

```diff
@@ -0,0 +1,48 @@
+import type { ReactNode } from 'react';
+
+interface DataTableProps {
+  /**
+   * Array of column header text
+   */
+  headers: string[];
+
+  /**
+   * 2D array of table data where each inner array represents a row
+   */
+  data: (string | number | ReactNode)[][];
```

Comment:
I think this could be confusing to developers. Which array is the inner array?

What if you did something like this:

```ts
type Cell = (string | number | ReactNode);
type Row = Cell[];
type Table = { /* rest */; data: Row[]; }
```

This disambiguates the structure so that it's clear how the data should be structured.

Reply:
Or even just
```ts
type Row = (string | number | ReactNode)[];
type Table = { data: Row[] }
```